File: blog-admin/src/api/friendLink.ts
import request from './request'
import { FriendLink } from '@/types'
import { FriendLinkReviewStatus, FriendLinkVisibilityStatus } from '@/types/enums'

export interface FriendLinkDTO {
  name: string
  url: string
  logo: string
  description: string
  author: string
  sort?: number
  reviewStatus?: FriendLinkReviewStatus
  isVisible?: FriendLinkVisibilityStatus
}

export const getFriendLinkList = (params?: {
  id?: number
  userId?: number
  name?: string
  author?: string
  email?: string
  url?: string
  reviewStatus?: FriendLinkReviewStatus
  isVisible?: FriendLinkVisibilityStatus
}): Promise<FriendLink[]> => {
  return request.get('/admin/friend-link/list', { params })
}

export const createFriendLink = (data: FriendLinkDTO): Promise<number> => {
  return request.post('/admin/friend-link', data)
}

export const updateFriendLink = (id: number, data: FriendLinkDTO): Promise<void> => {
  return request.put(`/admin/friend-link/${id}`, data)
}

export const deleteFriendLink = (id: number): Promise<void> => {
  return request.delete(`/admin/friend-link/${id}`)
}

export const approveFriendLink = (id: number): Promise<void> => {
  return request.post(`/admin/friend-link/${id}/approve`)
}

export const rejectFriendLink = (id: number): Promise<void> => {
  return request.post(`/admin/friend-link/${id}/reject`)
}

File: blog-service/src/main/java/com/blog/modules/friendlink/model/entity/FriendLink.java
package com.blog.modules.friendlink.model.entity;


import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;
import lombok.EqualsAndHashCode;
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("friend_link")
public class FriendLink extends com.blog.shared.model.entity.BaseEntity {

    @TableField("user_id")
    private Long userId;

    private String name;

    private String url;

    private String logo;

    private String description;

    private String author;

    private String email;

    @TableField("is_visible")
    private Integer isVisible;

    @TableField("review_status")
    private Integer reviewStatus;

    private Integer sort;
}

